src/main/java/com/sigopt/model/Subresource.java

```diff
@@ -0,0 +1,59 @@
+package com.sigopt.model;
+
+import com.sigopt.net.APIMethodCaller;
+import com.sigopt.net.PaginatedAPIMethodCaller;
+
+public class Subresource<T extends APIObject> extends BoundObject {
+    String name;
+    Class<T> klass;
+
+    public Subresource(String prefix, String name, Class<T> klass) {
+        super(prefix);
+        this.name = name;
+        this.klass = klass;
+    }
+
+    public APIMethodCaller<T> fetch() {
+        return new APIMethodCaller<T>("get", this.prefix() + "/" + this.name + "/:id", klass);
```

Comment:
does this work when `prefix` is empty, or does it lead to double slashes?

Reply:
https://github.com/sigopt/sigopt-java/blob/bde26035f5d662ab1754100be3638fd88fd2e0dd/src/main/java/com/sigopt/model/Experiment.java#L176

This is directly from the original, for subresources the prefix should never be empty by definition. There is a possibility of us adding extra slashes in the subresource but I don't think thats a huge concern and if it is its a separate one from the changes being made for TM

Reply:
sounds good